tests/integration/common-operator-integ-suite.sh

```diff
@@ -14,118 +14,208 @@
 set -eu -o pipefail
 
-if [ $# -eq 0 ]; then
-  echo "No arguments provided"
-  exit 1
-fi
+check_arguments() {
+  if [ $# -eq 0 ]; then
+    echo "No arguments provided"
+    exit 1
+  fi
+}
 
-while [ $# -gt 0 ]; do
-  case "$1" in
-    --ocp)
-      shift
-      OCP=true
-      ;;
-    --kind)
-      shift
-      OCP=false
-      ;;
-    *)
-      echo "Invalid flag: $1"
-      exit 1
-      ;;
-  esac
-done
-
-if [ "${OCP}" == "true" ]; then
-  echo "Running on OCP"
-else
-  echo "Running on kind"
-fi
+parse_flags() {
+  SKIP_BUILD=false
+  while [ $# -gt 0 ]; do
+    case "$1" in
+      --ocp)
+        shift
+        OCP=true
+        ;;
+      --kind)
+        shift
+        OCP=false
+        ;;
+      --skip-build)
+        shift
+        SKIP_BUILD=true
+        ;;
+      *)
+        echo "Invalid flag: $1"
+        exit 1
+        ;;
+    esac
+  done
+
+  if [ "${OCP}" == "true" ]; then
+    echo "Running on OCP"
+  else
+    echo "Running on kind"
+  fi
+}
 
-WD=$(dirname "$0")
-WD=$(cd "$WD" || exit; pwd)
+initialize_variables() {
+  WD=$(dirname "$0")
+  WD=$(cd "${WD}" || exit; pwd)
 
-NAMESPACE="${NAMESPACE:-istio-operator}"
-DEPLOYMENT_NAME="${DEPLOYMENT_NAME:-istio-operator}"
-CONTROL_PLANE_NS="${CONTROL_PLANE_NS:-istio-system}"
-COMMAND="kubectl"
+  NAMESPACE="${NAMESPACE:-istio-operator}"
+  DEPLOYMENT_NAME="${DEPLOYMENT_NAME:-istio-operator}"
+  CONTROL_PLANE_NS="${CONTROL_PLANE_NS:-istio-system}"
+  COMMAND="kubectl"
 
-if [ "${OCP}" == "true" ]; then
-  COMMAND="oc"
-fi
+  if [ "${OCP}" == "true" ]; then
+    COMMAND="oc"
+  fi
 
-BRANCH="${BRANCH:-maistra-3.0}"
+  echo "Using command: ${COMMAND}"
 
-if [ "${OCP}" == "true" ]; then
-  ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-openshift.yaml"
-else
-  ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-kubernetes.yaml"
-fi
+  if [ "${OCP}" == "true" ]; then
+    ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-openshift.yaml"
+  else
+    ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-kubernetes.yaml"
+  fi
 
-TIMEOUT="3m"
+  TIMEOUT="3m"
+}
 
-check_ready() {
-    local NS=$1
-    local POD_NAME=$2
-    local DEPLOYMENT_NAME=$3
+get_internal_registry() {
+  # Validate that the internal registry is running, configure the variable to be used in the Makefile. 
+  # If there is no internal registry, the test can't be executed targeting to the internal registry
 
-    echo "Check POD: NAMESPACE: \"${NS}\"   POD NAME: \"${POD_NAME}\""
+  # Check if the registry pods are running
+  ${COMMAND} get pods -n openshift-image-registry --no-headers | grep -v "Running" && echo "It looks like the OCP image registry is not deployed or Running. This tests scenario requires it. Aborting." && exit 1
+
+  # Check if default route already exist
+  if [ -z "$(${COMMAND} get route default-route -n openshift-image-registry -o name)" ]; then
+    echo "Route default-route does not exist, so you can perform patching here."
```

Comment:
This wording is confusing. Are you telling the user they can perform a patch here?

Reply:
My bad, changing the phrase